src/StatusLine.php

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * Interface representing a Value Object of the HTTP Status-Line, as
+ * specified in RFC 2616 and RFC 7231.
+ *
+ * PHP version 5.3
+ *
+ * @category StatusLine
+ *
+ * @package Teapot
+ *
+ * @author    Barney Hanlon <barney@shrikeh.net>
+ * @copyright 2013-2016 B Hanlon. All rights reserved.
+ * @license   MIT http://opensource.org/licenses/MIT
+ * @link      https://shrikeh.github.com/teapot
+ */
+namespace Teapot;
+
+use Teapot\StatusCodeException\InvalidStatusCodeException;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Interface representing a Value Object of the HTTP Status-Line, as
+ * specified in RFC 2616 and RFC 7231.
+ *
+ * PHP version 5.3
+ *
+ * @category StatusLine
+ *
+ * @package Teapot
+ *
+ * @author    Barney Hanlon <barney@shrikeh.net>
+ * @copyright 2013-2016 B Hanlon. All rights reserved.
+ * @license   MIT http://opensource.org/licenses/MIT
+ * @link      https://shrikeh.github.com/teapot
+ */
+class StatusLine
```

Comment:
@shrikeh 

I hear people have different feelings in regard to suffixing `Interface`, `Trait`, or prefixing `Abstract` when it comes to interfaces, classes, and abstractions.

However, since you've already extracted a `StatusCodeInterface` now, I would suggest to be consistent and use `StatusLineInterface`, what do you think?